### ISSUE TITLE
whatsits interfere with letterspacing

### ISSUE THREAD

**Issue by u-fischer** [u-fischer/luaotfload#53]
~~~~
\documentclass[]{article}

\usepackage{fontspec,xcolor}
\setmainfont{TeX Gyre Termes}[LetterSpace=30]

\begin{document}
{Auf\color{red}gabe}
\end{document}

~~~~

gives

![image](https://user-images.githubusercontent.com/4047173/55491245-6e949f00-5635-11e9-8cd8-31aa0787803c.png)

This is also imho the reason why letterspacing doesn't work well when selnolig is used, as selnolig inserts whatsits nodes to break the ligatures. 

https://tex.stackexchange.com/questions/482961/unable-to-supress-ligatures-in-headings-which-are-set-in-caps

**Comment by zauguin** [comment]
I tested the PR with the code from the StackExchange question: The "normal" selnolig version is fixed, the `\breaklig` version still does not have proper letterspacing. But `\breaklig` does not use whatsits, so that is expected.

**Comment by u-fischer** [comment]
Will be resolved in version 2.97.
